pkg/vsphere/provisioning/progress/progress.go

```diff
@@ -17,20 +17,31 @@ const (
 	pathPrefix            = "/api/vsphere/v1/provisioning/progress.json"
 	pollInterval          = 5 * time.Second
 	progressCompleteValue = 100
+
+	// StatusFailed indicates that the progress failed.
+	StatusFailed = -1
+	// StatusSuccess indicates that the progress succeeded.
+	StatusSuccess = 1
+	// StatusInProgress indicates that the progress is still ongoing.
+	StatusInProgress = 2
+	// StatusCancelled indicates that the progress has been cancelled.
+	StatusCancelled = 3
 )
 
 // Progress contains information regarding the provisioning of a VM returned by the API .
 type Progress struct {
 	// TaskIdentifier is the identifier of the provisioning task.
 	TaskIdentifier string `json:"identifier"`
-	// Queued indicated that the task is waiting to be started-
+	// Queued indicates that the task is waiting to be started.
 	Queued bool `json:"queued"`
 	// Progress is the provisioning progress in percent (queuing not included).
 	Progress int `json:"progress"`
 	// VMIdentifier of the new VM.
 	VMIdentifier string `json:"vm_identifier"`
 	// Errors encountered while provisioning.
 	Errors []string `json:"errors"`
+	// Status of the task.
+	Status int `json:"status"`
```

Comment:
It probably makes sense to use a dedicated type for this instead of an generic `int`.

Reply:
Yes, I thought about that too, but decided to keep it flexible and instead exposed the known statuses as constants.

I would put this change into the followup ticket mentioned for integration tests.

Reply:
We wouldn't lose any of the flexibility if we convert it to a proper type, or am I missing something here?